lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -2861,6 +2861,63 @@ bool FoldChannelShuffle::run(Function *F, const CompilationContext &cctx) {
   return changed;
 }
 
+// Fold Tile -> Add into BatchedAdd wherever applicable.
+bool FoldTileAddIntoBatchedAdd::run(Function *F,
+                                    const CompilationContext &cctx) {
+  LOG_SCOPE(F->getLogContext(), getName());
+
+  bool changed = false;
+  for (auto &node : F->getNodes()) {
```

Comment:
```suggestion
  for (const auto &node : F->getNodes()) {
```
I like `const`-ing things wherever possible. `node` and `addNode` are not mutated so I think this should be possible but I could be wrong.